lib/validation.rb

```diff
@@ -100,7 +100,7 @@ def self.validate_storage_volumes(storage_volumes, boot_disk_index)
     end
     storage_volumes.each { |volume|
       volume.each_key { |key|
-        fail ValidationFailed.new({storage_volumes: "Invalid key: #{key}"}) unless allowed_keys.include?(key)
+        fail ValidationFailed.new({storage_volumes: "Invalid key: #{key}"}) unless allowed_keys.include?(key.to_sym)
```

Comment:
Nice one, though I wonder if we should consider constraining the inputs, or if that is just too impractical (usually because keyword arguments or ruby program literals are easier as symbols, and sometimes we're working with data, which tends to come in as strings)

Reply:
Fair point. I updated the PR to explicitly convert keys of storage volume definition (that we read from the ai_models.yml) to symbols. And got rid of this commit.